pipes/WDL/tasks/tasks_assembly.wdl

```diff
@@ -181,6 +181,121 @@ task scaffold {
     }
 }
 
+task ivar_trim {
+    File    aligned_bam
+    File?   trim_coords_bed
+    Int?    min_keep_length
+
+    String? docker="andersenlabapps/ivar"
+
+    String  bam_basename=basename(aligned_bam, ".bam")
+
+    command {
+        set -ex -o pipefail
+        ivar version | tee VERSION
+        if [ -n "${trim_coords_bed}" ]; then
+          ivar trim -e -i ${aligned_bam} -b ${trim_coords_bed} -p trim ${'-m ' + min_keep_length}
+          samtools sort -@ `nproc` -m 1500M -o ${bam_basename}.trimmed.bam trim.bam
+        else
+          cp ${aligned_bam} ${bam_basename}.trimmed.bam
+        fi
+    }
+
+    output {
+        File   aligned_trimmed_bam = "${bam_basename}.trimmed.bam"
+        String ivar_version        = read_string("VERSION")
+    }
+
+    runtime {
+        docker: "${docker}"
+        memory: "7 GB"
+        cpu: 4
+        disks: "local-disk 375 LOCAL"
+        dx_instance_type: "mem1_ssd1_v2_x4"
+    }
+}
+
+task refine_assembly_with_aligned_reads {
+    File     reference_fasta
+    File     reads_aligned_bam
+    String   sample_name
+
+    Boolean? mark_duplicates=false
+    Float?   major_cutoff=0.5
+    Int?     min_coverage=2
+
+    String?  docker="quay.io/broadinstitute/viral-assemble"
+
+    command {
+        set -ex -o pipefail
+
+        # find 90% memory
+        mem_in_mb=`/opt/viral-ngs/source/docker/calc_mem.py mb 90`
+
+        assembly.py --version | tee VERSION
+
+        if [ ${true='true' false='false' mark_duplicates} == "true" ]; then
+          read_utils.py mkdup_picard \
+            ${reads_aligned_bam} \
+            temp_markdup.bam \
+            --JVMmemory "$mem_in_mb"m \
+            --loglevel=DEBUG
+        else
+          ln -s ${reads_aligned_bam} temp_markdup.bam
+        fi
+        samtools index -@ `nproc` temp_markdup.bam temp_markdup.bai
+
+        ln -s ${reference_fasta} assembly.fasta
+        assembly.py refine_assembly \
+          assembly.fasta \
+          temp_markdup.bam \
+          refined.fasta \
+          --already_realigned_bam=temp_markdup.bam \
+          --outVcf ${sample_name}.sites.vcf.gz \
+          --min_coverage ${min_coverage} \
+          --major_cutoff ${major_cutoff} \
+          --JVMmemory "$mem_in_mb"m \
```

Comment:
Not yet passing the novoalign license?

Reply:
ah -- I thought just git merging origin/master after you merged your last PR would've done the trick, but of course there's this!